pkg/operator/ceph/pool/peers.go

```diff
@@ -58,3 +62,76 @@ func (r *ReconcileCephBlockPool) reconcileAddBootstrapPeer(pool *cephv1.CephBloc
 
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileCephBlockPool) removeStaleClusterPeers(pool *cephv1.CephBlockPool) error {
+	mirrorInfo, err := client.GetPoolMirroringInfo(r.context, r.clusterInfo, pool.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
+	}
+
+	if len(mirrorInfo.Peers) == 0 {
+		logger.Debug("no stale peer cluster site is required to be removed.")
+		return nil
+	}
+
+	// if both the peers.SecretNames count in the spec and actual mirror peers are equal, then no stale peer site is available.
+	if len(pool.Spec.Mirroring.Peers.SecretNames) != len(mirrorInfo.Peers) {
+		if len(pool.Spec.Mirroring.Peers.SecretNames) == 0 {
+			for _, peer := range mirrorInfo.Peers {
+				err := client.RemoveClusterPeer(r.context, r.clusterInfo, pool.Name, peer.UUID)
```

Comment:
How about a helper method for lines 81-85? It looks repeated below on 95-99

Reply:
The implementation might change based on our discussion in the huddle today. I'll try disabling mirroring and removing peer sites for both `image` and `pool` mode. Currently we support only `pool` mode.

Reply:
marking this as resolved.